### PR TITLE
Feature/36431/update fake barts importer

### PR DESCRIPTION
**Pull request by shilpigoeldev**
## What?

~~Fake Barts~~ handler to import Lynch data for linking.

this has now been moved to NHS England Adhoc importer, but since we have imported it as Barts file on Live API at svn we still have it under `R1H`.

## How ?

Used Barts colorectal handler space to import Lynch data for linking patient data from the different sources.

## Testing?

Added relevant test case.

## Anything Else?

Since its a fake handler as per Fiona -" its the data linkage rather than the data format that we care about in this instance"

### CODE REVIEW

**Comment by shilpigoeldev** [comment]
@ollietulloch , after confirming with Fiona  have now changed provider to be NHS England 'X26' , and under it an adhoc handler with Fiona's comments explaining which data is it about. Does it look ok ? TIA 🙇‍♀️

File: db/migrate/20240620121939_add_nhs_england_provider.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddNhsEnglandProvider < ActiveRecord::Migration[6.1]
+  def change
+     Zprovider.create(zproviderid: 'X26') unless Zprovider.exists?('X26')
```
Comment (ollietulloch) [inline db/migrate/20240620121939_add_nhs_england_provider.rb:3]: Does this need a `shortdesc`?

Comment (ollietulloch) [reply db/migrate/20240620121939_add_nhs_england_provider.rb:3]: Oh, I see `shortdesc` isn't included in other migrations

Comment (shilpigoeldev) [reply db/migrate/20240620121939_add_nhs_england_provider.rb:3]: None of the past Zprovider additions have added 'shortdesc' , I suppose as its mainly staging database which finally ends up in CAS having proper Zprovider table.

File: test/lib/import/colorectal/providers/nhs_england/adhoc_handler_colorectal_test.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    @record = build_raw_record('pseudo_id1' => 'bob')
+    @genocolorectal = Import::Colorectal::Core::Genocolorectal.new(@record)
+    @importer_stdout, @importer_stderr = capture_io do
+      @handler = Import::Colorectal::Providers::NhsEngland::AdhocHandlerColorectal.new(EBatch.new)
+    end
+    @logger = Import::Log.get_logger
+  end
+
+  test 'assign_variant_details' do
+    fake_record = build_raw_record('pseudo_id1' => 'bob')
```
Comment (ollietulloch) [inline test/lib/import/colorectal/providers/nhs_england/adhoc_handler_colorectal_test.rb:14]: Not sure `fake_record` is a good name here?

Comment (shilpigoeldev) [reply test/lib/import/colorectal/providers/nhs_england/adhoc_handler_colorectal_test.rb]: Updated in latest commit 🙏

**Review by ollietulloch — APPROVED** [review]

**Pull request merged**